File: rust/poke-api/src/pokemon.rs
use serde::Serialize;

use crate::pokemon_specie::PokemonSpecie;

#[derive(Serialize)]
pub struct Pokemon {
    id: u32,
    name: String,
}

impl From<PokemonSpecie> for Pokemon {
    fn from(ps: PokemonSpecie) -> Self {
        let PokemonSpecie { id, name, .. } = ps;

        Self { id, name }
    }
}

File: typescript/poke-api/src/pokemon.ts
import type { PokemonSpecie } from "./pokemon_specie";

interface Pokemon {
	id: string;
	name: string;
	flavorText: string[];
}

export function transformPokemon(pokemonSpecie: PokemonSpecie): Pokemon {
	const flavorTextEntries = pokemonSpecie.flavorTextEntries.filter(
		(entry) => entry.language.name === "en",
	);

	const flavorText: string[] = [];
	for (const flavor of flavorTextEntries) {
		flavorText.push(flavor.flavorText);
	}

	const pokemon = {
		id: pokemonSpecie.id,
		name: pokemonSpecie.name,
		flavorText,
	};

	return pokemon;
}
